src/components/Modals/FiatRamps/views/Overview.tsx

```diff
@@ -121,17 +133,65 @@ export const Overview: React.FC<OverviewProps> = ({
   chainId,
   setChainId,
   chainAdapterManager,
+  handleAccountIdChange,
+  accountId,
 }) => {
   const translate = useTranslate()
   const { fiatRampAction } = useParams<{ fiatRampAction: FiatRampAction }>()
   const toast = useToast()
   const { fiatRamps } = useModal()
-
-  const [shownOnDisplay, setShownOnDisplay] = useState<Boolean | null>(null)
-
   const {
     state: { wallet },
   } = useWallet()
+  const multiAccountsEnabled = useFeatureFlag('MultiAccounts')
+  const [shownOnDisplay, setShownOnDisplay] = useState<Boolean | null>(null)
+  const accountBalances = useAppSelector(selectPortfolioAccountBalances)
+  const marketData = useAppSelector(state =>
+    selectMarketDataById(state, selectedAsset?.assetId ?? ''),
+  )
+  const [accountAddress, setAccountAddress] = useState<string | null>(null)
+
+  // TODO: change the following with `selectPortfolioAccountMetadataByAccountId`
+  // once web/#2632 got merged
+  const accountMeta = useAppSelector(selectPortfolioAccountMetadata)[accountId ?? '']
+  const accountType = accountMeta?.accountType
+  const bip44Params = accountMeta?.bip44Params
+  useEffect(() => {
+    if (!(wallet && bip44Params && selectedAsset)) return
+    ;(async () => {
+      const { chainId } = fromAssetId(selectedAsset.assetId)
+      const chainAdapter = chainAdapterManager.get(chainId)!
+      const selectedAccountAddress = await chainAdapter.getAddress({
+        wallet,
+        accountType,
+        bip44Params,
+      })
+      setAccountAddress(selectedAccountAddress)
+    })()
+  }, [accountType, wallet, bip44Params, selectedAsset, chainAdapterManager])
+  const minimumSellThreshold = useMemo(
+    () => bnOrZero(supportedFiatRamps[fiatRampProvider].minimumSellThreshold ?? 0),
+    [fiatRampProvider],
+  )
+  const hasEnoughBalance = useMemo(
+    () =>
+      bnOrZero(
+        multiAccountsEnabled
+          ? (accountId && accountBalances[accountId][selectedAsset?.assetId ?? '']) ?? 0
```

Comment:
If we're using `bnOrZero` we shouldn't need to optional chain to zero. This will evaluate to `undefined` in the unhappy path, and will be effectively defaulted to zero.